### DISCUSSION TITLE
Does refreshing a session work with OIDC providers?

### DISCUSSION THREAD

**Discussion by akkie** [ory/kratos#2723] [Q&A] [answered]
Hi,

I have implemented the session refresh as described in: https://www.ory.sh/docs/kratos/guides/setting-up-cors#refreshing-a-session

Does this method work with OIDC logins? For me it doesn't work. Logging-in with username/email and password refreshes the session in my case. But if I try to login with an OIDC provider, it doesn't work.

The DB entry for the flow has the `request_url` with value `https://auth.my.app/self-service/login/browser?refresh=true&return_to=https%3A%2F%2Fmy.app%2Fsettings%2Faccount` and the `active_method` is `oidc` after the login. But the session was not refreshed. The redirect to the return_to path does also not work. It redirects to the default redirect URL.

Log-in itself with OIDC connect works as expected. 

The question is, should this normally work or is this not implemented?

**Comment by akkie** [comment, marked as answer]
I have found the issue. I use the JS SDK and the `initializeSelfServiceLoginFlowForBrowsers` function has no cookies parameter. But it seems that it needs it. After submitting the cookie with the following method, the refreshing works now as expected.

```

kratosApi.initializeSelfServiceLoginFlowForBrowsers(refresh, aal, returnTo, {
  headers: { cookies }
})
```

**Reply by aeneasr** [reply]
Nice find! We are also adjusting the SDK so that passing cookie becomes a first class citizen
